### PR TITLE
[2.0] Update preprocessor symbols and convert to table layout

### LINKED ISSUE #2561: [4.7/4.7.1/2.0] Update preprocessor symbols
This covers an existing need for 4.7, but 4.7 with 4.7.1+2.0 can perhaps can be updated on the same pass.
```
.NET Framework 4.7   --> NET47
.NET Framework 4.7.1 --> NET471
.NET Standard 2.0    --> NETSTANDARD2_0
```

e.g. ...

*docs/core/tutorials/libraries.md*
https://docs.microsoft.com/en-us/dotnet/core/tutorials/libraries#how-to-target-the-net-framework

### PR DESCRIPTION
**Pull request by guardrex**
Adds preprocessor symbols for 2.0. See if a table format is better for these.

Fixes #2561

### CODE REVIEW

**Review by rpetrusha — APPROVED** [review]
I think that this is a vast improvement, @GuardRex. I'll approve the PR and leave it to you to merge if you also think it looks better.

**Comment by GuardRex** [comment]
Are we waiting until 2.0 ships to merge the ones that contain 2.0 content?

**Comment by GuardRex** [comment]
@rpetrusha @mairaw I think this is problematic after all. The table looks nice; but due to padding in the rows, the table is *insanely tall*. You can see where this is headed: It might pass today, but adding even just a few more symbols is going to overwhelm these topics where the table is used. (See screenshot below.)

Conversely, going back to that horrible code block format isn't an attractive option imo. Therefore, let me try something a little different here on the next commit. I have an idea that might work.

capture

**Comment by GuardRex** [comment]
@mairaw @rpetrusha Might be on to something here. How about [doing it this way](https://review.docs.microsoft.com/en-us/dotnet/core/tutorials/libraries?branch=pr-en-us-2744)*, which hardens the content against a crazy long list of symbols going forward?

*External reviewers: The image below reflects how the lastest commit renders on the review site, which isn't available publicly.

capture

**Comment by mairaw** [comment]
Not sure about this. I find the new table hard to parse.

**Comment by GuardRex** [comment]
> hard to parse

hard to understand?

**Comment by GuardRex** [comment]
@mairaw I *do* have another idear that might work. Stand-by for another version.

**Comment by GuardRex** [comment]
@mairaw This version attempts to explain without all the \<> \[] techo-babble.

Even if this doesn't work, I have even another idea to list them; but unlike the first version with one symbol per row, we can put the symbols into one cell of the table. I still think that will get unwieldy over time as more and more symbols come onto the stage.

**Comment by GuardRex** [comment]
Here's a shot showing how the latest version renders. The only other idea I've had is to list them all out in each of the three table cells (to avoid the mega-table of one symbol per row), but I still think that over time that if they're listed out even in three table cells that they'll get out of control at some point. See if this version works better than the last ...

capture

**Comment by rpetrusha** [comment]
@GuardRex, I think what developers want is a list of valid values, not an explanation of what's needed to form a valid value.

**Comment by GuardRex** [comment]
@rpetrusha Probably best then to go ahead and list them out but use one cell for each list to keep the table's height under control.

File: includes/preprocessor-symbols.md [hunk 1]
```diff
@@ -1,21 +1,5 @@
...
-.NET Standard 1.5         --> NETSTANDARD1_5
-.NET Standard 1.6         --> NETSTANDARD1_6
-.NET Core Application 1.0 --> NETCOREAPP1_0
-.NET Core Application 1.1 --> NETCOREAPP1_1
-```
+| Target Frameworks          | Symbol |
+| -------------------------- | ------ |
+| .NET Framework 2.0-4.7.1   | `NET20`, `NET35`, `NET40`, `NET45`, `NET451`, `NET452`, `NET46`, `NET461`, `NET462`, `NET47`, `NET471` |
+| .NET Standard 1.0-1.6, 2.0 | `NETSTANDARD1_0`, `NETSTANDARD1_1`, `NETSTANDARD1_2`, `NETSTANDARD1_3`, `NETSTANDARD1_4`, `NETSTANDARD1_5`, `NETSTANDARD1_6`, `NETSTANDARD2_0` |
+| .NET Core Application 1.0-1.1, 2.0 | `NETCOREAPP1_0`, `NETCOREAPP1_1`, `NETCOREAPP2_0`). |
```
Comment (svick) [inline includes/preprocessor-symbols.md]: The stray `).` at the end should be removed.

**Comment by GuardRex** [comment]
This looks nice ... @mairaw u like this approach?

capture

**Comment by GuardRex** [comment]
One thing tho.... need to change header to "Symbols" (plural).

**Comment by rpetrusha** [comment]
@Guardrex, I think that this looks good. There is, though, one inconsistency in the TFM section that it would be nice to fix. Two formats are used for TFMs that have multiple equivalents:  `[alt1] [alt2]`, and `[alt1, alt2]`. It would be good to standardize on  `[alt1] [alt2]`, and so change `netcore45 [win, win8]` to `netcore45 [win] [win8]`.

**Comment by GuardRex** [comment]
@rpetrusha This PR is just for the preprocessor symbol table.

**Comment by GuardRex** [comment]
@rpetrusha Patched here: https://github.com/dotnet/docs/pull/2809

File: includes/preprocessor-symbols.md [hunk 2]
```diff
@@ -1,21 +1,5 @@
...
-.NET Standard 1.3         --> NETSTANDARD1_3
-.NET Standard 1.4         --> NETSTANDARD1_4
-.NET Standard 1.5         --> NETSTANDARD1_5
-.NET Standard 1.6         --> NETSTANDARD1_6
-.NET Core Application 1.0 --> NETCOREAPP1_0
-.NET Core Application 1.1 --> NETCOREAPP1_1
-```
+| Target Frameworks          | Symbols |
+| -------------------------- | ------- |
+| .NET Framework 2.0-4.7.1   | `NET20`, `NET35`, `NET40`, `NET45`, `NET451`, `NET452`, `NET46`, `NET461`, `NET462`, `NET47`, `NET471` |
```
Comment (mairaw) [inline includes/preprocessor-symbols.md]: I like the new format. Should we just call it .NET Framework, .NET Standard and .NET Core without version or application on the first column?

**Review by mairaw — APPROVED** [review]
